src/security/KeyData.cc

```diff
@@ -97,33 +98,36 @@ Security::KeyData::loadX509ChainFromFile()
         return;
     }
 
-#if TLS_CHAIN_NO_SELFSIGNED // ignore self-signed certs in the chain
     if (SelfSigned(*cert)) {
-        debugs(83, DBG_PARSE_NOTE(2), "Certificate is self-signed, will not be chained: " << *cert);
-    } else
-#endif
-    {
+        debugs(83, DBG_PARSE_NOTE(2), "Signing certificate is self-signed: " << *cert);
```

Comment:
The original (and previously disabled) "Certificate is self-signed, will not be chained: <signing certificate>" debugs() message contents sounded odd because the signing certificates should never be added to the chain object (signing certificates are sent because they are added to the TLS context object), regardless of whether they are self-signed. What we probably meant to say is that Squid will not form/sent a chain of CA certificates after this signing certificate because/when the signing certificate is self-signed. When the TODO below is implemented, we will warn about those "extra" certificates (if any).

Reply:
FTR; when seen in the context of cache.log trace with the surrounding certificate load sequence, this looks much more correct. It will only look "odd" if the file contains a misconfiguration - in which case the oddity makes it clearer to admin where/which misconfiguration occured.